app/src/main/java/com/example/bucketbuds/FriendsAdapter.java

```diff
@@ -0,0 +1,188 @@
+            }
+        }
+
+        public void bind(User user, Boolean friendsBool){
+            if (friendsBool) {
+                UserPub otherUserPub = user.getUserPub();
+                tvBucketCount.setText(String.valueOf(otherUserPub.getBucketCount()));
+                tvFriendCount.setText(String.valueOf(otherUserPub.getFriendCount()));
+                // user selects a friend to remove
+                View.OnClickListener removeFriendClickListener = new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        otherUserPub.removeFriend(currentUser);
+                        currentUserPub.removeFriend(user);
+                        users.remove(user);
+                        updateFriends(false, getAdapterPosition(), otherUserPub);
+                    }
+                };
+                ivRemoveFriend.setOnClickListener(removeFriendClickListener);
+            } else {
+                // user is already added as a friend but appears in search
+                if (friendsId.contains(user.getObjectId())){
+                    ivAddFriend.setVisibility(View.GONE);
+                } else {
+                    ivAddFriend.setVisibility(View.VISIBLE);
+                    // user selects a friend to add
+                    View.OnClickListener addFriendClickListener = new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            UserPub otherUserPub = user.getUserPub();
+                            otherUserPub.addFriend(currentUser);
+                            currentUserPub.addFriend(user);
+                            ivAddFriend.setVisibility(View.GONE);
+                            updateFriends(true, getAdapterPosition(), otherUserPub);
+                        }
+                    };
+                    ivAddFriend.setOnClickListener(addFriendClickListener);
+                }
+            }
+            if (user.getImage() != null) {
+                Glide.with(context).load(user.getImage().getUrl()).circleCrop().into(ivProfileImage);
+            } else {
+                Glide.with(context).load(R.drawable.profile_placeholder).circleCrop().into(ivProfileImage);
+            }
+            tvUsername.setText("@"+user.getUsername());
+            tvFirstName.setText(user.getFirstName());
+            tvLastName.setText(user.getLastName());
+        }
+    }
+
+    public void clear(){
+        users.clear();
+    }
+
+    // Helper method for saving added or removed friend
+    public void updateFriends(Boolean added, int position, UserPub otherUserPub){
+
+        SaveCallback otherUserSaveCallback = new SaveCallback() {
+            @Override
+            public void done(ParseException e) {
+                if (e != null) {
+                    Toast.makeText(context, "Friend action failed", Toast.LENGTH_SHORT).show();
```

Comment:
suggestion: Consider a `Log.e` here that prints the `ParseException` received for easier debugging.